aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java

```diff
@@ -512,6 +516,9 @@ public Connection waitForConnectionReady() {
                 return new Connection("Thread interrupted waiting for connection acknowledgement");
             }
             LOG.debug("Current endpoint status: " + endpointStatus.get());
+            if(EndpointStatus.CONNECTION_FAILED.equals(endpointStatus.get())) {
+                return new Connection("Connection failed.");
+            }
```

Comment:
For the following use case:
 - WiFi is off.
 - Client attempts to save a mutation.

This change ensures that we fail immediately, instead of waiting 10 seconds for the start_ack.